components/public-api-server/pkg/apiv1/oidc_test.go

```diff
@@ -405,12 +447,35 @@ func setupOIDCService(t *testing.T, expClient experiments.Client) (*protocol.Moc
 
 	_, handler := v1connect.NewOIDCServiceHandler(svc, connect.WithInterceptors(auth.NewServerInterceptor()))
 
-	srv := httptest.NewServer(handler)
-	t.Cleanup(srv.Close)
+	router := chi.NewRouter()
+	router.Use(middleware.Logger)
+	router.Mount("/", handler)
+	ts := httptest.NewServer(router)
+	t.Cleanup(ts.Close)
 
-	client := v1connect.NewOIDCServiceClient(http.DefaultClient, srv.URL, connect.WithInterceptors(
+	client := v1connect.NewOIDCServiceClient(http.DefaultClient, ts.URL, connect.WithInterceptors(
 		auth.NewClientInterceptor("auth-token"),
 	))
 
-	return serverMock, client, dbConn, srv.URL
+	return serverMock, client, dbConn
+}
+
+func newFakeIdP(t *testing.T, enableDiscovery bool) string {
+	router := chi.NewRouter()
+	ts := httptest.NewServer(router)
+	t.Cleanup(ts.Close)
+	url := ts.URL
+
+	router.Use(middleware.Logger)
+	if enableDiscovery == true {
+		router.Get("/.well-known/openid-configuration", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
If you wanted a richer mock, have a look at `stripe_test.go` where we use `go-vcr` which can record, and replay fixtures from real providers, providing more realistic tests

Reply:
Thanks for the hint. Here, it's really about the existence of the endpoint.